### ISSUE TITLE
Async DNS Library

### ISSUE THREAD

**Issue by jots** [johnjansen/crystal-libraries-needed#18]
like (ruby) https://github.com/alexdalitz/dnsruby
or (golang): https://github.com/miekg/dns

**Comment by RX14** [comment]
There's a very basic "DNS" client for looking up names in the stdlib already, although it just binds gataddrinfo.

**Comment by jots** [comment]
yes I've used that one, but need async for resolving millions of names in short amount of time, and also ability to specify nameservers to hit for resolution on a per query basis.

**Comment by RX14** [comment]
@jots It's already async; just spawn a new fiber for each query, however it's still not suitable for more than looking up hostnames' A records I agree.

**Comment by stakach** [comment]
https://github.com/spider-gazelle/dns

**Comment by paulocoghi** [comment]
If everyone agrees, I believe this can be closed. And thanks for @stakach again! ❤️
